src/share/redolog/ob_log_disk_manager.cpp

```diff
@@ -1965,25 +1965,33 @@ int ObLogDiskManager::get_total_disk_space(int64_t& total_space) const
 
 int ObLogDiskManager::get_total_disk_space_(int64_t& total_space) const
 {
-  int ret = OB_EAGAIN;
-  struct statfs fsst;
-  // try every disk until succeed
-  for (int32_t i = 0; OB_FAIL(ret) && i < MAX_DISK_COUNT; i++) {
-    if (OB_LDS_GOOD == disk_slots_[i].get_state()) {
-      if (0 != ::statfs(disk_slots_[i].get_disk_path(), &fsst)) {
-        ret = OB_IO_ERROR;
-        COMMON_LOG(WARN, "statfs error", K(ret), K(errno), KERRMSG);
-      } else {
-        ret = OB_SUCCESS;
-        total_space = (int64_t)fsst.f_bsize * (int64_t)fsst.f_blocks;
+  int ret = OB_SUCCESS;
+  const int64_t clog_disk_limit_size = ObServerConfig::get_instance().clog_disk_limit_size;
+
+  if (clog_disk_limit_size != 0){
+    total_space = clog_disk_limit_size;
```

Comment:
total_space should be less than disk total, min(clog_disk_limit_size, disk total)

Reply:
if clog_disk_limit_size set by user is greater than disk size, observer can't start normally. This check logic is done by "ObConfigClogDiskLimitSizeChecker::check" (in [src/share/config/ob_config_helper.cpp]).
So, clog_disk_limit_size must be less than disk total if observer can start and run normally.